app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt

```diff
@@ -532,33 +533,33 @@ class UploadWorker(
     }
 
     private fun findUniqueFileName(fileName: String): String {
-        var sequenceFileName: String?
-        var sequenceNumber = 1
-        while (true) {
+        var sequenceFileName: String? = fileName
+        val chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+        val random = Random()
+
+        // Loops until sequenceFileName does not match any existing file names
+        while (mediaClient
+                .checkPageExistsUsingTitle(
+                    String.format(
+                        "File:%s",
+                        sequenceFileName,
+                    ),
+                ).blockingGet()) {
+
+            // Generate a random 5-character alphanumeric string
+            val randomHash = (1..3)
+                .map { chars[random.nextInt(chars.length)] }
+                .joinToString("")
```

Comment:
Wow you are fast!

Would you mind keeping the number sequence strategy?

We are open to a hash strategy, but could you create a GitHub issue to discuss its pros and cons?

Thanks a lot! 🙂

Reply:
The issue I’m addressing #2506 does have discussion of the different strategies for speeding up this search and from what I could see a hash was voted for as the fastest and easiest to implement. I could revert to the number sequence but the code would remain virtually unchanged. The other option discussed was binary search to maintain the number sequence format of titles but improve search efficiency.

Reply:
Ah understood thanks! :-)
Sorry for the confusion I thought it was about https://github.com/commons-app/apps-android-commons/issues/5876